### PR TITLE
cmd/atlas/internal/migrate: template based migrate status

### PR DESCRIPTION
**Pull request by masseelch**
```
$ atlas migrate status --env local 
Migration Status: PENDING
  -- Current Version: 2 (1 statements applied)
  -- Next Version:    2 (1 statements left)
  -- Executed Files:  2 (last one partially)
  -- Pending Files:   2

Last migration attempt had errors:
  -- SQL:   asd bleh
  -- ERROR: near "asd": syntax error in a long line  with linebreaks
```

```
$ atlas migrate status --env local --log '{{ table . }}'
+---------+-------------+---------+-------+---------------------------+----------------+--------------------------+----------+
| VERSION | DESCRIPTION | STATUS  | COUNT |        EXECUTED AT        | EXECUTION TIME |          ERROR           |   SQL    |
+---------+-------------+---------+-------+---------------------------+----------------+--------------------------+----------+
|       1 |             | applied | 1/1   | 2022-09-28 10:51:01 +0200 | 1.85774ms      |                          |          |
+---------+-------------+---------+-------+---------------------------+----------------+--------------------------+----------+
|       2 |             | applied | 1/2   | 2022-09-28 10:51:04 +0200 | 1.139969ms     | near "asd": syntax error | asd bleh |
+         +-------------+---------+-------+---------------------------+----------------+--------------------------+----------+
|         |             | pending | 1/2   |                           |                |                          |          |
+---------+-------------+---------+-------+---------------------------+----------------+--------------------------+----------+
|       3 |             | pending |       |                           |                |                          |          |
+---------+-------------+---------+-------+---------------------------+----------------+--------------------------+----------+
```

```
$ atlas migrate status --env local --log '{{ json . "  " }}'
{
  "Env": {
    "Driver": "sqlite3",
    "URL": {
      "Scheme": "sqlite",
      "Opaque": "",
      "User": null,
      "Host": "",
      "Path": "/home/jcl/Code/ariga/atlas/build/file.db",
      "RawPath": "",
      "OmitHost": false,
      "ForceQuery": false,
      "RawQuery": "",
      "Fragment": "",
      "RawFragment": "",
      "DSN": "/home/jcl/Code/ariga/atlas/build/file.db",
      "Schema": "main"
    },
    "Dir": "/home/jcl/Code/ariga/atlas/build/migrations"
  },
  "Available": [
    {
      "Name": "1.sql",
      "Version": "1"
    },
    {
      "Name": "2.sql",
      "Version": "2"
    },
    {
      "Name": "3.sql",
      "Version": "3"
    }
  ],
  "Pending": [
    {
      "Name": "2.sql",
      "Version": "2"
    },
    {
      "Name": "3.sql",
      "Version": "3"
    }
  ],
  "Applied": [
    {
      "Version": "1",
      "Description": "",
      "Type": "applied",
      "Applied": 1,
      "Total": 1,
      "ExecutedAt": "2022-09-28T10:51:01.125445262+02:00",
      "ExecutionTime": 1857740,
      "OperatorVersion": "Atlas CLI - development"
    },
    {
      "Version": "2",
      "Description": "",
      "Type": "applied",
      "Applied": 1,
      "Total": 2,
      "ExecutedAt": "2022-09-28T10:51:04.117059587+02:00",
      "ExecutionTime": 1139969,
      "Error": "near \"asd\": syntax error in a long line\n\nwith linebreaks",
      "ErrorStmt": "asd bleh",
      "OperatorVersion": "Atlas CLI - development"
    }
  ],
  "Current": "2",
  "Next": "2",
  "Count": 1,
  "Total": 2,
  "Status": "PENDING",
  "Error": "near \"asd\": syntax error in a long line  with linebreaks",
  "SQL": "asd bleh"
}

```

### CODE REVIEW

File: cmd/atlas/internal/migrate/ent/schema/revision.go [hunk 1]
```diff
@@ -46,6 +46,8 @@ func (Revision) Fields() []ent.Field {
 			GoType(time.Duration(0)),
 		field.Text("error").
 			Optional(),
+		field.Text("error_stmt").
+			Optional(),
```
[the remark this answers is not in the document]
Comment (masseelch) [reply cmd/atlas/internal/migrate/ent/schema/revision.go:50]: Yes

File: cmd/atlas/internal/cmdapi/migrate.go [hunk 2]
```diff
@@ -85,6 +85,9 @@ var (
 			GitDir  string // repository working dir
 			GitBase string // branch name to compare with
 		}
+		status struct {
+			Format string // log formatting
```
[the remark this answers is not in the document]
Comment (masseelch) [reply cmd/atlas/internal/cmdapi/migrate.go:89]: 2 lines above the same comment, I kept it for clarity as above

File: cmd/atlas/internal/migrate/run.go [hunk 3]
```diff
@@ -0,0 +1,289 @@
+// Copyright 2021-present The Atlas Authors. All rights reserved.
```
Comment (a8m) [inline cmd/atlas/internal/migrate/run.go:1]: Change filename. e.g. report.go

File: cmd/atlas/internal/migrate/run_test.go [hunk 4]
```diff
@@ -0,0 +1,116 @@
+// Copyright 2021-present The Atlas Authors. All rights reserved.
+// This source code is licensed under the Apache 2.0 license found
+// in the LICENSE file in the root directory of this source tree.
+
+package migrate
```
Comment (a8m) [inline cmd/atlas/internal/migrate/run_test.go:5]: Ditto

**Review by a8m — APPROVED** [review]

**Pull request merged**